packages/authorization-process/test/addClientPurpose.test.ts

```diff
@@ -840,4 +842,333 @@ describe("addClientPurpose", async () => {
       purposeAlreadyLinkedToClient(mockPurpose.id, mockClient.id)
     );
   });
+  it("should write on event-store for the addition of a purpose into a client where the requester is the delegate", async () => {
```

Comment:
If I understood correctly, even when the purpose is delegated, the original consumer can still add it to a client, so I would add a test like `"should write on event-store for the addition of a purpose created by a delegate into a client where the requester is the Consumer"`